File: 237_Delete_Node_in_a_Linked_List/Java/Solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) { val = x; }
 * }
 */
public class Solution {
    public void deleteNode(ListNode node) {
        if(node.next != null) {
            node.val = node.next.val;
            if(node.next.next != null) {
                node.next = node.next.next;    
            }
            else {
                node.next = null;
            }
        }
        else {
            // If it's a single node
            node = null;
            node.next = null;
        }
    }
}

File: 237_Delete_Node_in_a_Linked_List/CSharp/Solution.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace LeetCode_237_Delete_Node_in_a_Linked_List
{
    public class Solution
    {
        public void DeleteNode(ListNode node)
        {
            if (null == node.next)
            {
            }
            else
            {
                node.val = node.next.val;
                node.next = node.next.next;
            }
        }
    }
}
